gap/measure_contention.g

```diff
@@ -26,55 +23,80 @@ GAUSS_prepare := function(nrAvailableThreads)
     fi;
 end;
 
-GAUSS_compute := function(A, q, numberChops)
-    return GAUSS_GET_REAL_TIME_OF_FUNCTION_CALL(Chief,
+GAUSS_compute := function(A, q, numberChops, showOutput)
+    if (showOutput) then
+        Print("Starting the parallel execution.\n");
+    fi;
+    res := GAUSS_GET_REAL_TIME_OF_FUNCTION_CALL(Chief,
                                             [GF(q), A, numberChops, numberChops, true]);
+    # The variable `time` stores the cpu time that was used by the execution
+    # of the last statement.
+    res.cpuTime := time;
+    return res;
 end;
 
-GAUSS_evaluate := function(nrAvailableThreads, bench, A, visible)
+GAUSS_evaluate := function(nrAvailableThreads, bench, A, showOutput)
     local CPUTimeCompute, resPar, benchStd, resStd, correct, counters,
     totalAcquired, totalContended, factor;
 
-    CPUTimeCompute := time;
     resPar := bench.result;
-    benchStd := GAUSS_GET_REAL_TIME_OF_FUNCTION_CALL(EchelonMatTransformation, [A]);
-    resStd := benchStd.result;
-    correct := -1 * resStd.vectors = resPar.vectors
-           and -1 * resStd.coeffs = resPar.coeffs;
-    if not correct then
-        ErrorNoReturn("GaussPar: Result incorrect!");
+    # Use ms instead of microseconds
+    bench.time := Round(bench.time / 1000.);
```

Comment:
One can change the value of the input of a function?

Reply:
Or did I overlook something here?

Reply:
Ah, GAP evaluates the argument before passing it to Round. So it computes `bench.time / 1000` first, and then passes the result to Round.